cli/azd/pkg/executil/executil.go

```diff
@@ -197,10 +198,31 @@ type RunArgs struct {
 	EnrichError bool
 }
 
+type RedactData struct {
+	matchString   *regexp.Regexp
+	replaceString string
+}
+
 func (rr RunResult) String() string {
 	return fmt.Sprintf("exit code: %d, stdout: %s, stderr: %s", rr.ExitCode, rr.Stdout, rr.Stderr)
 }
 
+func RedactSensitiveData(msg string) string {
+	regexpRedact := make(map[string]RedactData)
```

Comment:
consider a name change here like:

```suggestion
	regexpRedactRules := make(map[string]RedactData)
```